src/controller/ControllerPositionHierarchy.cpp

```diff
@@ -40,5 +40,30 @@ namespace UKControllerPlugin {
                     return position == storedPosition.get();
             }) != this->positions.cend();
         }
+
+        bool ControllerPositionHierarchy::operator==(const ControllerPositionHierarchy& compare) const
+        {
+            if (compare.CountPositions() != this->CountPositions()) {
+                return false;
+            }
+
+            std::vector<std::reference_wrapper<const ControllerPosition>>::const_iterator it1
+                = this->positions.cbegin();
+            std::vector<std::reference_wrapper<const ControllerPosition>>::const_iterator it2
+                = compare.positions.cbegin();
+
+            while (it1 != this->positions.cend())

```

Comment:
This loop could be a little more readable ;)

Reply:
Added a comment to make it more meaningful :)